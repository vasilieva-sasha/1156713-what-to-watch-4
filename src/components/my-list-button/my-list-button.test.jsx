import React from "react";
import renderer from "react-test-renderer";
import configureStore from 'redux-mock-store';
import {Provider} from "react-redux";
import MyListButton from "./my-list-button";
import NameSpace from './../../reducer/name-space';

const mock = [{
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  genre: `Drama`,
  releaseDate: 2018,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  posterInfo: `posters/placeimg_218_327_animals`,
  background: `background/placeimg_1000_424_animals`,
  backgroundColor: ``,
  preview: ``,
  rating: {
    score: 8.9,
    level: `Very good`,
    count: 240
  },
  text: `text`,
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: true
},
{
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  genre: `Drama`,
  releaseDate: 2018,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  posterInfo: `posters/placeimg_218_327_animals`,
  background: `background/placeimg_1000_424_animals`,
  backgroundColor: ``,
  preview: ``,
  rating: {
    score: 8.9,
    level: `Very good`,
    count: 240
  },
  text: `text`,
  director: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  isFavorite: false
}];

const mockStore = configureStore([]);
const store = mockStore({
  [NameSpace.DATA]: {
    films: mock
  },
  [NameSpace.USER]: {
    authorizationStatus: `AUTH`
  }
});

describe(`MyListButtonComponent`, () => {
  it(`MyListButtonComponent in list`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <MyListButton film={mock[0]} handleChangeFavoriteStatus={() => {}} authorizationStatus={store.authorizationStatus}/>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`MyListButtonComponent not in list`, () => {
    const tree = renderer.create(
        <Provider store={store}>
          <MyListButton film={mock[1]} handleChangeFavoriteStatus={() => {}} authorizationStatus={store.authorizationStatus}/>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
