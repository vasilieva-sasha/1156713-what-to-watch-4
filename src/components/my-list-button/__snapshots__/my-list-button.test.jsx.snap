// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListButtonComponent MyListButtonComponent in list 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#in-list"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;

exports[`MyListButtonComponent MyListButtonComponent not in list 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;
